tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -2766,6 +2766,23 @@ void DeclResultIdMapper::storeToShaderOutputVariable(
     const StageVarDataBundle &stageVarData) {
   SpirvInstruction *ptr = varInstr;
 
+  // Since boolean output stage variables are represented as unsigned
+  // integers, we must cast the value to uint before storing.
+  if (isBooleanStageIOVar(stageVarData.decl, stageVarData.type,
+                          stageVarData.semantic->getKind(),
+                          stageVarData.sigPoint->GetKind())) {
+    QualType finalType = varInstr->getAstResultType();
+    if (stageVarData.arraySize != 0) {
+      // We assume that we will only have to write to a single value of the
```

Comment:
This depends on how you define "single value". In our example, `position` would end of having a type of `bool[arraySize][2]`. It would still be true that we will only need to write a single `bool[2]` to loation `position[invocationId]`. The single value is the entire `bool[2]`.

The code is correct, if you think the comment needs improvement please suggest something else. I'm okay with something being 90% true if it gets the idea across succinctly.

Reply:
Discusses this with Steven 'offline', it's a similar but unrelated bug I was looking at (https://godbolt.org/z/oz5no5f44). Fine for this PR.